reframe/frontend/statistics.py

```diff
@@ -70,6 +72,75 @@ def retry_report(self):
 
         return '\n'.join(report)
 
+    def output_dict(self):
+        report = []
+        current_run = rt.runtime().current_run
+        for t in self.tasks(current_run):
+            check = t.check
+            partition = check.current_partition
+            partname = partition.fullname if partition else 'None'
+            environ_name = (check.current_environ.name
+                            if check.current_environ else 'None')
+            nodelist = (','.join(check.job.nodelist)
+                        if check.job and check.job.nodelist else '<None>')
+            job_type = 'local' if check.is_local() else 'batch job'
+            jobid = check.job.jobid if check.job else -1
+            if t.failed:
+                result = 'fail'
+            else:
+                result = 'success'
+
+            report.append({
+                'test': check.name,
```

Comment:
```suggestion
                'name': check.name,
```
I would put `name` as the key, but the `test` key is also fine if you prefer it.